src/olympia/addons/models.py

```diff
@@ -1542,21 +1542,24 @@ def get_lookup_field(cls, identifier):
                 lookup_field = 'slug'
         return lookup_field
 
+    @property
+    def addonguid__guid(self):
```

Comment:
I get the idea being the name, but I find it confusing when used in a queryset like you do below...

Why do we need this anyway? Don't we have a guarantee nowadays than `self.addonguid` is going to exist ? Couldn't we use _that_ directly everywhere instead of having the property?

Reply:
If Addon.guid is `None` it wouldn't get created (that would be invalid, and it's unlikely to ever affect blocklist code because it's all based on guid).  Plus there isn't an absolute guarantee that every Addon will have a matching AddonGUID in the database because the fk is on AddonGUID rather than on Addon.

Changing it to be `self.addonguid.guid` directly is more straightforward, at the potential risk of a traceback somewhere in the future.

Reply:
Although maybe the name conflicting with the queryset syntax is an accident waiting to happen.

Reply:
Yeah I think we need a) that explanation or a variant of it in a comment or docstring and b) a new name to avoid the queryset syntax.

Reply:
https://github.com/mozilla/addons-server/pull/14469/commits/ba4ba1778b9729cf3c290a116f35e2e11bf90b0c